storage/operation/commits.go

```diff
@@ -17,13 +17,13 @@ import (
 //
 // CAUTION:
 //   - Confirming that no value is already stored and the subsequent write must be atomic to prevent data corruption.
-//     The caller must acquire the [storage.LockInsertOwnReceipt] and hold it until the database write has been committed.
+//     The caller must acquire the [storage.LockIndexStateCommitment] and hold it until the database write has been committed.
 //
 // Expected error returns during normal operations:
 //   - [storage.ErrDataMismatch] if a *different* state commitment is already indexed for the same block ID
 func IndexStateCommitment(lctx lockctx.Proof, rw storage.ReaderBatchWriter, blockID flow.Identifier, commit flow.StateCommitment) error {
-	if !lctx.HoldsLock(storage.LockInsertOwnReceipt) {
-		return fmt.Errorf("cannot index state commitment without holding lock %s", storage.LockInsertOwnReceipt)
+	if !lctx.HoldsLock(storage.LockIndexStateCommitment) {
+		return fmt.Errorf("cannot index state commitment without holding lock %s", storage.LockIndexStateCommitment)
 	}
```

Comment:
What do you think about a utility function that packages together the lock check and error construction, to make sure the error message always matches the lock we are checking?

```go
// MustHoldLock checks that a proof holds the given lock.
// If it does, returns nil.
// If it does not, returns a generic exception indicating which lock was not held.
func MustHoldLock(proof lockctx.Proof, lockID string) error {
	if proof.HoldsLock(lockID) {
		return nil
	}
	return fmt.Errorf("missing required lock: %s", lockID)
}
```

Then we can do:
```suggestion
	if err := storage.MustHoldLock(lctx, storage.LockIndexStateCommitment); err != nil {
		return fmt.Errorf("cannot index state commitment: %w", err)
	}
```